index.rst

```diff
@@ -2,24 +2,209 @@
+The idea relies on continuously deploying and testing (ideally in an automated way) new and existing software with a well defined and understood set of unit and integration tests.
+Each new piece of functionality contributes to ultimately fulfilling a high-level requirement, where the verification is continually performed and maintained via the testing frameworks.
 
-Abstract
-========
+Although a large portion of features/requirements can be tested/verified in a CI/CD approach, there are some cases where customized tests are necessary.
+This includes, but is not limited to, tests that are directly related to hardware or requirements that are indirectly related to software, e.g. hexapod axis positioning or telescope mount tracking precision.
+
+We also provide artifacts that can be used to certify that the requirements were successfully verified.
+
+Here, we detail the various strategies and tools developed to support software requirement verification for Telescope and Site Software.
+
+Introduction
+============
+
+Requirement verification is an important process in the delivery of any project and is what ensures that a project completes its intended deliverables.
+In other words, requirement verification provides answers to the question
+
+> Have we built what we intended to build?
+
+Because software is, by definition, extremely dynamic, requirements verification can become quite challenging.
+The Telescope and Site Software team, as most software teams in industry at large, utilize an Agile methodology, which means we are continually adapting to focus on the most useful features and delivering new functionality to users at a high cadence.
+This also means we are constantly receiving feedback from users and making adjustments to the software based on those feedback.
+
+In terms of requirements verification, the fact that software is constantly changing means we need to be continually verifying requirements.
+In order to achieve a constant stream of new features and implement a user-feedback mechanism, we rely on Continuous Integration and Continuous Delivery (CI/CD).
+
+The CI/CD discipline originates from the `Agile methodology <https://www.agilealliance.org/agile101/>`__ as a way to respond to the rapid delivery of new software on a project.
+Overall, the idea is that each new software feature (or bug-fixes) must also contains a set of tests that verify the functionality.
+In general, tests are separated in two different categories; unit tests and integration tests.
+
+Unit tests are small scale tests designed to verify a particular functionality of a software component.
+They usually run in isolation utilizing mocks for other components when required.
+Unit tests are the lowest level of testing and, although they usually exercise the requirements, they are not sufficient for requirement verification.
+This is often because the are executed in isolated environments or there is no generated report which can be easily evaluated by a reviewer.
```

Comment:
Oh, this phrase is referring to the unit tests.. I know rob generate some reports for SAL unit tests but the majority of our unit testing uses pytest, which one could argue also generates a report, but I don't think it would be suitable for a requirement verification artifact.